rllib/algorithms/dqn/dqn_rainbow_catalog.py

```diff
@@ -0,0 +1,281 @@
+        return TorchCategorical
+
+    def _get_encoder_config(
+        self,
+    ) -> ModelConfig:
+        """Returns the encoder config.
+
+        In case noisy networks should be used, the base encoder config
+        (`self._encoder_config`) is ovverriden and a `NoisyMLPEncoderConfig`
+        is used.
+
+        Returns:
+            If noisy networks should be used a `NoisyMLPEncoderConfig` else
+            `self._encoder_config` defined by the parent class.
+        """
+        # Check, if we use
+        use_lstm = self._model_config_dict["use_lstm"]
+        use_attention = self._model_config_dict["use_attention"]
+
+        # In cases of LSTM or Attention, fall back to the basic encoder.
+        if not use_lstm and not use_attention:
+            # Check, if the observation space is 1D Box. Only then we can use an MLP.
+            if (
+                isinstance(self.observation_space, Box)
+                and len(self.observation_space.shape) == 1
+            ):
+                # Define the encoder hiddens.
+                if self._model_config_dict["encoder_latent_dim"]:
+                    af_and_vf_encoder_hiddens = self._model_config_dict["fcnet_hiddens"]
+                else:
+                    af_and_vf_encoder_hiddens = self._model_config_dict[
+                        "fcnet_hiddens"
+                    ][:-1]
+
+                # Instead of a regular MLP use a NoisyMLP.
+                return NoisyMLPEncoderConfig(
+                    input_dims=self.observation_space.shape,
+                    hidden_layer_dims=af_and_vf_encoder_hiddens,
+                    hidden_layer_activation=self._model_config_dict["fcnet_activation"],
+                    hidden_layer_use_layernorm=self._model_config_dict[
+                        "hidden_layer_use_layernorm"
+                    ],
+                    hidden_layer_use_bias=self._model_config_dict[
+                        "hidden_layer_use_bias"
+                    ],
+                    hidden_layer_weights_initializer=self._model_config_dict[
+                        "fcnet_weights_initializer"
+                    ],
+                    hidden_layer_weights_initializer_config=self._model_config_dict[
+                        "fcnet_weights_initializer_config"
+                    ],
+                    hidden_layer_bias_initializer=self._model_config_dict[
+                        "fcnet_bias_initializer"
+                    ],
+                    hidden_layer_bias_initializer_config=self._model_config_dict[
+                        "fcnet_bias_initializer_config"
+                    ],
+                    output_layer_activation=self.af_and_vf_encoder_activation,
+                    output_layer_dim=self.latent_dims[0],
+                    output_layer_use_bias=self._model_config_dict[
+                        "output_layer_use_bias"
+                    ],
+                    # TODO (sven, simon): Should these initializers rather the fcnet
```

Comment:
@sven1977 Do we really use here in the encoder already the `post_fcnet` settings or should these only go into the heads? I see this also in the default settings in the `core/models/base.py` and the other new stack algorithms. Let's make a decision here and go with it from here.

Reply:
I would think so.
Let's say the encoder is a 3 layer CNN with an additional [512, 512] head on top (after flattening the last CNN layer's output).
Then we need the post_fcnet_hiddens here to set up these last 2 512-layers.

Then each head (Af, Vf) can still have its own config with m `fcnet_hiddens`.